src/averageAges.js

```diff
@@ -55,7 +66,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => people.some(human =>
+    human.mother === person.name));
+
+  const children = onlyWithSon ? people.filter(person =>
+    person.mother !== null && person.sex === 'm')
+    : people.filter(person => person.mother !== null);
+
+  const ageDifferences = [];
+
+  children.forEach(child => {
+    const searchedMother = mothers.find(mother =>
+      mother.name === child.mother);
+
+    if (searchedMother) {
+      ageDifferences.push(child.born - searchedMother.born);
+    }
+  });
```

Comment:
use map if you want to return an array
```suggestion
  const ageDifferences = children.map(child => {
```